presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkTaskRdd.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import org.apache.spark.Partition;
+import org.apache.spark.SparkContext;
+import org.apache.spark.TaskContext;
+import org.apache.spark.rdd.RDD;
+import org.apache.spark.rdd.ZippedPartitionsBaseRDD;
+import org.apache.spark.rdd.ZippedPartitionsPartition;
+import scala.Tuple2;
+import scala.collection.Seq;
+import scala.reflect.ClassTag;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyIterator;
+import static java.util.Objects.requireNonNull;
+import static scala.collection.JavaConversions.asJavaIterator;
+import static scala.collection.JavaConversions.asScalaBuffer;
+import static scala.collection.JavaConversions.asScalaIterator;
+import static scala.collection.JavaConversions.seqAsJavaList;
+
+/**
+ * PrestoSparkTaskRdd represents execution of Presto stage, it contains:
+ * - A list of shuffleInputRdds, each of the corresponding to a child stage.
+ * - An optional taskSourceRdd, which represents ALL table scan inputs in this stage.
+ * <p>
+ * Table scan is present when joining a bucketed table with an unbucketed table, for example:
+ * Join
+ * /  \
+ * Scan  Remote Source
+ * <p>
+ * In this case, bucket to Spark partition mapping has to be consistent with the Spark shuffle partition.
+ * <p>
+ * shuffleInputRdds can also be empty when the stage partitioning is SINGLE_DISTRIBUTION.
```

Comment:
Maybe emphasize both `taskSourceRdd` and `shuffleInputRdds` will be empty when the stage partitioning is `SINGLE_DISTRIBUTION` ?

Reply:
Changed the comment to 

```
 * When the stage partitioning is SINGLE_DISTRIBUTION and the shuffleInputRdds is empty,
 * the taskSourceRdd is expected to be present and contain exactly one empty partition.
```